src/dmd/dtemplate.d

```diff
@@ -1351,8 +1351,37 @@ extern (C++) final class TemplateDeclaration : ScopeDsymbol
             {
                 Parameter fparam = fparameters[parami];
 
+                /* https://issues.dlang.org/show_bug.cgi?id=16486
+                 * If 'prmtype' is template alias, resolve it,
+                 * to the instance it's aliased to before proceeding.
+                 */
+                Type prmtype = fparam.type;
+                //printf("prmtype: %s\n", prmtype.toChars());
+                while (prmtype.ty == Tinstance)
+                {
+                    TypeInstance prmti = cast(TypeInstance) prmtype;
+                    //printf("prmti: %s\n", prmti.toChars());
+                    if (prmti.tempinst.findTempDecl(sc, null))
```

Comment:
Please transform all of this nesting if to 
```
if (!cond)
    continue;
```
So that the code tays at the same level of indentation.